File: leetcode2/second/0179. Largest Number.py
class Solution:
    def largestNumber(self, nums: 'List[int]') -> 'str':
        if sum(nums)==0:
            return '0'
        
        nums=self.subsort(nums)
        return ''.join(str(num) for num in nums)
        
    def subsort(self,nums):
        for i in range(len(nums)):
            for j in range(i,len(nums)):
                if self.cmp(nums[i],nums[j]):
                    nums[i],nums[j]=nums[j],nums[i]
        return nums
    
    def cmp(self,a,b):
        return int(str(b)+str(a))>int(str(a)+str(b))

File: archives/C#/0179. Largest Number.cs
public class Solution {
    public string LargestNumber(int[] nums) {
        Array.Sort(nums,CompareInt);
        string str="";
        foreach(int num in nums){
            str+=num.ToString();
        }
        if(str[0]=='0')
            return "0";
        return str;
    }
    
    public int CompareInt(int num1,int num2){
        if (Convert.ToInt64(num1.ToString()+num2.ToString())-Convert.ToInt64(num2.ToString()+num1.ToString())>0)
            return -1;
        else
            return 1;
    }
}
